iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2149,12 +2149,16 @@ public List<Partition> getPartitions(org.apache.hadoop.hive.ql.metadata.Table hm
   }
 
   public boolean isPartitioned(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
-    if ((hmsTable.getAsOfVersion() != null || hmsTable.getAsOfTimestamp() != null) &&
-            hasUndergonePartitionEvolution(hmsTable) ||
-        !hmsTable.getTTable().isSetId()) {
+    if (!hmsTable.getTTable().isSetId()) {
       return false;
     }
     Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
+    Snapshot snapshot = IcebergTableUtil.getTableSnapshot(table, hmsTable);
+    boolean isTimeTravel = snapshot != null && table.currentSnapshot() != null &&
```

Comment:
could we use `equals` here?
````
  public boolean isPartitioned(org.apache.hadoop.hive.ql.metadata.Table hmsTable) {
    if (!hmsTable.getTTable().isSetId()) {
      return false;
    }
    Table table = IcebergTableUtil.getTable(conf, hmsTable.getTTable());
    Snapshot snapshot = IcebergTableUtil.getTableSnapshot(table, hmsTable);

    boolean isTimeTravel = snapshot != null && snapshot.equals(table.currentSnapshot());
    if (isTimeTravel && hasUndergonePartitionEvolution(hmsTable)) {
      return false;
    }
    return table.spec().isPartitioned();
  }
````

Reply:
I agree that it is simpler and tolerant with unexpected snapshots, such as having the same snapshot id but different timestamps.
https://github.com/apache/iceberg/blob/c338323b2e9cd8a862fdf328ceacc1b59ea6275a/core/src/main/java/org/apache/iceberg/BaseSnapshot.java#L320-L332

I updated it. I kept the variable name, isTimeTravel, assuming reading a different snapshot using tag/branch sounds like a time-travel query.

Reply:
forked branch could be ahead of main, so I wouldn't call it time-travel + you can't rollback to a branch/tag

Reply:
I grabbed your point. I applied a slight modification to the variable name.
https://github.com/apache/hive/pull/5748/commits/1ac16b2ae6880228fcb76acb520e51b1d1b74306